File: scheduler/ssh_test.go
// Copyright 2023 The Yock Authors. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

package scheduler

import (
	"testing"
)

func TestSSH(t *testing.T) {
	sh, _ := NewSSHClient(SSHOpt{
		User:     "ubuntu",
		Pwd:      "root",
		IP:       "192.168.127.128",
		Redirect: true,
		Network:  "tcp",
	})

	sh.Put("../yock.tar", "release.tar")
	sh.Get("myfile.txt", "../myfile.txt")
}

File: lib/test/ssh_test.lua
ssh({
    user = "ubuntu",
    pwd = "root",
    ip = "192.168.127.128",
    port = 22,
    network = "tcp",
    redirect = true,
}, function(client)
    print(client:OS())
    -- client:Shell()
    -- client:Exec({ "echo 'root' | sudo -S ls" })
end)

-- s:Get("release.tar", "a.tar")
-- s:Put("../../yock.tar", "yock.tar")
